packages/firestore/src/api/credentials.ts

```diff
@@ -155,7 +156,9 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
     this.userCounter = 0;
 
     // Will fire at least once where we set this.currentUser
-    this.app.INTERNAL.addAuthTokenListener(this.tokenListener);
+    (this.app as _FirebaseApp).INTERNAL.addAuthTokenListener(
```

Comment:
Can you change app to be typed as _FirebaseApp and do the cast once at assignment time?

Reply:
I opted to do it case-by-case as we don't need to access the `INTERNAL` namespace often regardless. Ultimately it will be going away so I'd rather leave the individual casts just as "REFACTOR THIS" flags 😄

Reply:
The `INTERNAL` namespace was used for code sharing. We are going to do something formal in this area to remove that need. It's not immediate, but soon.